src/views/ScopesetComparison/index.jsx

```diff
@@ -0,0 +1,181 @@
+import { hot } from 'react-hot-loader';
+import { Component, Fragment } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { equals } from 'ramda';
+import { scopeUnion, scopeIntersection } from 'taskcluster-lib-scopes';
+import CodeEditor from '@mozilla-frontend-infra/components/CodeEditor';
+import Tooltip from '@material-ui/core/Tooltip';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import ScaleBalanceIcon from 'mdi-react/ScaleBalanceIcon';
+import Dashboard from '../../components/Dashboard/index';
+import Button from '../../components/Button';
+import splitLines from '../../utils/splitLines';
+
+@hot(module)
+@withStyles(theme => ({
+  actionButton: {
+    ...theme.mixins.fab,
+  },
+  listItemButton: {
+    ...theme.mixins.listItemButton,
+  },
+  redCell: {
+    backgroundColor: 'rgba(255, 0, 0, 0.25)',
+  },
+  greenCell: {
+    backgroundColor: 'rgba(0, 255, 0, 0.25)',
+  },
+  yellowCell: {
+    backgroundColor: 'rgba(255, 255, 0, 0.25)',
+  },
+  editorGrid: {
+    marginBottom: theme.spacing.unit,
+  },
+  cellGrid: {
+    padding: theme.spacing.unit,
+  },
+}))
+export default class ScopesetComparison extends Component {
+  state = {
+    scopeTextA: '',
+    scopeTextB: '',
+  };
+
+  handleScopesAChange = scopeTextA => {
+    this.setState({ scopeTextA });
+  };
+
+  handleScopesBChange = scopeTextB => {
+    this.setState({ scopeTextB });
+  };
+
+  handleCompareScopesClick = async () => {
+    const { scopeTextA, scopeTextB } = this.state;
+
+    if (scopeTextA && scopeTextB) {
+      const scopesA = splitLines(scopeTextA);
+      const scopesB = splitLines(scopeTextB);
+      const scopesetDiff = this.getScopesetDiff(scopesA, scopesB);
+      const cellColors = this.getCellColors(scopesetDiff);
+
+      this.setState({ scopesetDiff, cellColors });
+    }
+  };
+
+  getScopesetDiff = (scopesA, scopesB) => {
+    const scopesUnion = scopeUnion(scopesA, scopesB);
+    const scopesetDiff = [];
+
+    scopesUnion.forEach(scope => {
+      const s1 = scopeIntersection([scope], scopesA);
+      const s2 = scopeIntersection([scope], scopesB);
+
+      scopesetDiff.push([s1, s2]);
+    });
+
+    return scopesetDiff;
+  };
+
+  getCellColors = scopesetDiff => {
+    const cellColors = [];
+    const CELLS = {
+      G: 'green',
+      R: 'red',
+      Y: 'yellow',
+    };
+
+    scopesetDiff.forEach(scope => {
+      if (scope[0].length === 0 && scope[1].length) {
+        cellColors.push(['', `${CELLS.G}Cell`]);
```

Comment:
Why the use of `CELLS.`?  `${CELLS.G}` is longer than `green` and requires me as a reader to try to figure out what CELLS means..

Reply:
hm, I think I may have proposed that idea. @hemakshis feel free to change it to what Dustin is recommending :)